src/dev/i18n/extract_react_messages.test.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+*/
+
+import { parse } from '@babel/parser';
+import { isCallExpression, isJSXOpeningElement, isJSXIdentifier } from '@babel/types';
+
+import { extractIntlMessages, extractFormattedMessages } from './extract_react_messages';
+import { traverseNodes } from './utils';
+
+const intlFormatMessageCallSource = `
+const MyComponentContent = ({ intl }) => (
+  <input
+    type="text"
+    placeholder={intl.formatMessage({
+      id: 'message-id-1',
+      defaultMessage: 'Default message 1',
+      context: 'Message context 1'
+    })}
+  />
+);
+`;
+
+const formattedMessageElementSource = `
+class Component extends PureComponent {
+  render() {
+    return (
+      <p>
+        <FormattedMessage
+          id="message-id-2"
+          defaultMessage="Default message 2"
+          context="Message context 2"
+        />
+      </p>
+    );
+  }
+}
+`;
+
+const intlFormatMessageCallErrorSources = [
+  `
+const messageId = 'message-id'
+intl.formatMessage({
+  id: messageId,
+  defaultMessage: 'Default message',
+  context: 'Message context'
+});
+`,
+  `
+  const message = 'Default message'
+  intl.formatMessage({
+    id: 'message-id',
+    defaultMessage: message,
+    context: 'Message context'
+  });
+`,
+  `
+const context = 'Message context'
+intl.formatMessage({
+  id: 'message-id',
+  defaultMessage: 'Default message',
+  context
+});
+`,
+];
+
+describe('dev/i18n/extract_react_messages', () => {
+  describe('extractIntlMessages', () => {
+    it('extracts messages from "intl.formatMessage" function call', () => {
+      const expected = [
+        'message-id-1',
+        {
+          message: 'Default message 1',
+          context: 'Message context 1',
+        },
+      ];
+
+      const ast = parse(intlFormatMessageCallSource, { plugins: ['jsx'] });
+
+      let actual;
+
+      for (const node of traverseNodes(ast.program.body)) {
```

Comment:
__nit:__ I believe you don't have to worry about lazy iterator in tests, so something like this would be much easier to read:
```js
const expressionNode = [...traverseNodes(ast.program.body)].find(
  node => isCallExpression(node)
);
expect(extractIntlMessages(expressionNode)).toEqual(expected);
```